spec/services/versioned_files_service/object_spec.rb

```diff
@@ -40,8 +40,8 @@
     end
 
     context 'when the version manifest does not exist' do
-      it 'raises UnknownVersionError' do
-        expect { service.head_version }.to raise_error(VersionedFilesService::UnknowVersionError, 'Head version not found')
+      it 'returns 0' do
```

Comment:
Returning 0 doesn't seem like a clear way of indicating that there is no version.

Reply:
It's not clear to me that purl-fetcher needs to care whether it had an already published version or not. It seems like it just needs to know whether the version being offered is newer than what's already there.

Reply:
I've pushed up `090752a` that uses `nil` instead of `0`. I'm not sure having to scatter nil-checks around is a great improvement 🤷‍♂️

Reply:
Why not leave it as raising an UnknownVersionError?

Reply:
I guess not having a head version doesn't seem very exceptional (or particularly problematic) to me.

I think having `nil` or some nil-like value is much clearer than a method that throws an exception and having to remember to also check `head_version?` every time you want to access `head_version` (so you can avoid throwing an exception).